source/fundamentals/context.txt

```diff
@@ -36,6 +36,51 @@ an error.
     defer cancel()
     client.Database("<db>").Collection("<collection>").Insert(ctx, bson.D{{"x",1}})
 
+Single Timeout Setting
+----------------------
+
+You can set a single ``Timeout`` option on your ``Client`` to govern the
+amount of time that a single operation can execute using the
+``SetTimeout()`` method or specifying the ``timeoutMS`` option in your
+connection URI string. ``Database``, ``Collection``,
+``Session``, ``ChangeStream``, and ``Bucket`` instances elsewhere in
+your code inherit the ``Timeout`` option from ``Client`` if you do not set a
+Context for operations against the same entity. 
+
+If you pass a Context into an operation with a deadline, the driver uses
+that Context deadline for the operation. If the context does not have a
+deadline, the driver derives a new Context from the given Context using
+the ``Timeout`` option set on the ``Client``.
+
+The following code shows how to set the ``Timeout`` option on a ``Client``
+with the ``SetTimeout`` option:
+
+.. code-block:: go
+
+   opts := options.Client().SetTimeout(5 * time.Second)
+
+The following example shows how you can set a single timeout with the
+URI option and execute an operation that inherits this setting:
+
+.. code-block:: go
+
+   uri := "mongodb://user:pass@sample.host:27017/?timeoutMS=5000"
+   client := mongo.Connect(uri)
+   coll := client.Database("myDB").Collection("myColl")
+   ...
+   coll.InsertOne(context.Background(), doc)
+
+To learn more about URI options, see the :ref:`Connection Guide
+<golang-connection-options>`.
+
+.. important:: Legacy Timeout Options
+
+   ``SocketTimeout``, ``wTimeout``, ``MaxTime``, and ``MaxCommitTime``
+   will be deprecated in an upcoming release. The driver will ignore ``MaxTime`` and
```

Comment:
[clarification question] Just to double check, is the driver ignoring `MaxTime` and `MaxCommitTime` already, or is that the expected behavior once they're deprecated?

Reply:
Once all of the tickets in [this epic](https://jira.mongodb.org/browse/DRIVERS-555) are resolved, then the feature will be documented as stable according to the design doc. The Go Driver API docs say "NOTE(benjirewis): SetTimeout represents unstable, provisional API. The behavior of the driver when a Timeout is specified is subject to change."